### DISCUSSION TITLE
No traceroute command, but ‘pip install’ says it’s already present

### DISCUSSION THREAD

**Discussion by CablingOptional** [holzschu/a-shell#586] [Q&A] [answered]
Hello -

I tried to run traceroute, but I see the command isn’t available.
However, when I try to install it via ‘pip install traceroute’, I get a response which seems to indicate it’s already installed…so…I’m not sure what’s up lol

[~/Documents]$ traceroute
traceroute: command not found

[~/Documents]$ pip install traceroute
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: traceroute in /var/mobile/Containers/Data/Application/19529B3D-95AF-451D-92B8-D62D19D0280E/Library/lib/python3.11/sit
e-packages (19.2.0)

**Comment by holzschu** [comment, marked as answer]
- traceroute (and all the commands that provide the same functionality) require super-user privilege in order to function. That is completely impossible for an app on the AppStore. Roger Wolff (the maintainer of mtr) has a very nice explanation about this: https://github.com/traviscross/mtr/issues/204#issuecomment-723961118
- `pip install traceroute` only installs the Python package "traceroute", not the binary command itself.

**Reply by CablingOptional** [reply]
Ahh, okay - that makes sense - I appreciate the thorough explanation
